_posts/2022-07-06-how-to-use-airflow-to-schedule-trino-jobs.md

```diff
@@ -0,0 +1,376 @@
+---
+layout: post
+title:  "How to use Airflow with Trino"
+author: "Willie Valdez" 
+excerpt_separator: <!--more-->
+---
+
+The recent addition of the [fault-tolerant
+execution](/docs/current/admin/fault-tolerant-execution.html) architecture,
+delivered to Trino by Project Tardigrade, makes the use of Trino for running
+your ETL workloads an even more compelling alternative than ever before. We've
+set up a demo environment for you to easily give it a try in [Starburst
+Galaxy](https://www.starburst.io/platform/starburst-galaxy/).
+
+<!--more-->
+
+With Project Tardigrade providing an out-of-the-box solution with advanced
+resource-aware task scheduling and granular retries at the query level, we still
+need a robust tool to schedule and manage workloads themselves and Apache
+Airflow is a great choice for this purpose.
+
+[Apache Airflow](https://airflow.apache.org/) is a widely used workflow engine that allows you to schedule and
+run complex data pipelines. Airflow provides many plug-and-play operators and
+hooks to integrate with many third-party services like Trino.
+
+To get started using Airflow to run data pipelines with Trino you need to
+complete the following steps:
+
+* Install of Apache Airflow 2.10+
+* Install the TrinoHook
+* Create a Trino connection in Airflow
+* Deploy a TrinoOperator
+* Deploy your DAGs
+
+# Installing Apache Airflow in Docker
+
+The best way to get you going, if you don’t already have an Airflow cluster
+available, is to run Airflow in a container using docker compose. Just be
+aware that this is not best practice for a production environment.
+
+Requirements for the host:
+
+* Docker 
+* Docker Compose 1.28+
+
+Step 1) Create a directory named airflow for all our configuration files.
+
+```
+$ mkdir airflow
+```
+
+Step 2) In the airflow directory create three subdirectory called `dags`, `plugins`, and `logs`.
+
+```
+$ cd airflow
+$ mkdir dags plugins logs
+```
+
+Step 3) Download the Airflow docker compose yaml file.
+
+```
+$ curl -LfO 'https://airflow.apache.org/docs/apache-airflow/stable/docker-compose.yaml'
+```
+
+Step 4) Create an `.env` configuration file: 
+
+```
+$ echo -e "AIRFLOW_UID=$(id -u)" > .env
+$ echo "AIRFLOW_GID=0" >> .env 
+```
+
+Step 5) Start the Airflow containers
+
+```
+$ docker-compose up -d
+```
+
+# Installing the TrinoHook
+
+The best practice, if running Airflow in docker, is to install the TrinoHook in
```

Comment:
I think we need to actually change this to say that you must do this .. it is not best practice.. its required for things to work